src/test/java/org/jetbrains/kotlinx/lincheck/test/strategy/modelchecking/ObstructionFreedomViolationTest.kt

```diff
@@ -0,0 +1,51 @@
+/*-
+ * #%L
+ * Lincheck
+ * %%
+ * Copyright (C) 2019 - 2020 JetBrains s.r.o.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.jetbrains.kotlinx.lincheck.test.strategy.modelchecking
+
+import org.jetbrains.kotlinx.lincheck.*
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.strategy.*
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.*
+import org.jetbrains.kotlinx.lincheck.verifier.*
+import org.junit.*
+
+class ObstructionFreedomViolationTest : VerifierState() {
```

Comment:
That is strange, because I get
```
= Obstruction-freedom is required but a lock has been found =
Execution scenario (init part):
[incAndGet(), incAndGet(), get(), incAndGet(), incAndGet()]
Execution scenario (parallel part):
| get()       | incAndGet() |
| get()       | incAndGet() |
| incAndGet() | get()       |
| get()       | get()       |
| incAndGet() | incAndGet() |
Execution scenario (post part):
[incAndGet(), incAndGet(), get(), incAndGet(), get()]
= The following interleaving leads to the error =
Parallel part interleaving:
|                                                                                               | incAndGet()                                                                                         |
|                                                                                               |   MONITOR ENTER at ObstructionFreedomViolationTest.incAndGet(ObstructionFreedomViolationTest.kt:36)  |
|                                                                                               |   c.READ: 4 at ObstructionFreedomViolationTest.incAndGet(ObstructionFreedomViolationTest.kt:36)     |
|                                                                                               |   c.WRITE(5) at ObstructionFreedomViolationTest.incAndGet(ObstructionFreedomViolationTest.kt:36)    |
|                                                                                               |   switch                                                                                            |
| get()                                                                                         |                                                                                                     |
|   MONITOR ENTER at ObstructionFreedomViolationTest.get(ObstructionFreedomViolationTest.kt:39) |                                                                                                     |
|                                                                                               |   MONITOR EXIT at ObstructionFreedomViolationTest.incAndGet(ObstructionFreedomViolationTest.kt:36)  |

```